File: 6_kyu/Esolang_Interpreters_1_Introduction_to_Esolangs_and_My_First_Interpreter_MiniStringFuck.rs
fn my_first_interpreter(code: &str) -> String {
    let mut output = String::new();
    let mut accu = 0_u8;
    for c in code.chars() {
        match c {
            '+' => accu = accu.wrapping_add(1),
            '.' => output.push(accu as char),
            _ => {} // ignore invalid code
        }
    }
    output
}

File: 6_kyu/Esolang_Interpreters_1_Introduction_to_Esolangs_and_My_First_Interpreter_MiniStringFuck.lua
-- MiniStringFuck interpreter.
--    https://esolangs.org/wiki/MiniStringFuck
-- Argument:
--    Code to execute.
--    Chars could be + (increment action) or . (print action).
--    All other chars in the string are ignored.
-- Return:
--    Printed characters in a string.
function myFirstInterpreter(code)
  local accu = ""
  local cell = 0
  for plusdot in code:gmatch("%+*%.?") do
    if plusdot:sub(-1) == '.' then
      cell = (cell+#plusdot-1) % 256
      accu = accu .. string.char(cell)
    else
      cell = (cell+#plusdot) % 256
    end 
  end
	return accu
end

return myFirstInterpreter
